### DISCUSSION TITLE
Mac keycode for Do Not Disturb (and other keycodes?)

### DISCUSSION THREAD

**Discussion by rainymatch** [kmonad/kmonad#646] [Q&A]
Hey everybody! I was trying to find the keycode for the "Do Not Disturb" key (F6 on the newer macbooks) and failed. In the "apple" keymap here there are only the older keycodes for these special mac keys (affecting keyboard backlight) in the keycode.hs (https://github.com/kmonad/kmonad/blob/master/src/KMonad/Keyboard/Keycode.hs ) I did not find it either. Is there possibly some number of the key (like the infamous mac "[Cannot translate from mac keycode: (7,100)]"? If somebody could elaborate shortly on this topic so that I can find my keycodes next time on my own, I would indeed appreciate that a lot!

**Comment by postylem** [comment]
I think Apple doesn't make this available, unfortunately, so it's not possible, likewise for the 'dictation' button that newer macs have.  But I may be wrong.

**Reply by tim-phillips** [reply]
Dictation key and others are defined in:
https://github.com/kmonad/kmonad/blob/606c488862df6c3be1e4e670f6e1e06008c84732/src/KMonad/Keyboard/Keycode.hs#L411-L415
but no dnd key.
